.githooks/pre-commit

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# File to check (relative to repo root)
+FILENAME="ip_country/src/dbip_country.rs"
+REMOTE=$(git rev-parse --abbrev-ref --symbolic-full-name @{u})
+
+# Exit early if file is not part of the commit
+if ! git diff --cached --name-only | grep -q "^$FILENAME$"; then
+  exit 0
+fi
+
+# Fetch latest remote version of the file
+git fetch origin > /dev/null 2>&1
+
+# Get remote file blob SHA
+REMOTE_BLOB=$(git ls-tree "$REMOTE" "$FILENAME" | awk '{print $3}')
+
+if [ -z "$REMOTE_BLOB" ]; then
+  echo "⚠️  Remote version of $FILENAME not found in $REMOTE."
+  exit 1
+fi
+
+# Get remote file size
+REMOTE_SIZE=$(git cat-file -s "$REMOTE_BLOB")
+
+# Get local file size (staged version, not working copy)
+LOCAL_SIZE=$(git ls-files --stage | grep "$FILENAME" | awk '{print $2}' | xargs git cat-file -s)
+
+if [ "$LOCAL_SIZE" != "$REMOTE_SIZE" ]; then
```

Comment:
Now that you have that `("NOEX", "Nonexistent")` country in the small file, you could find out whether the local `dbip_country.rs` file is the small one or the generated one by `grep`ping through it for that string.

Make sure the error message explains what's being searched for, though, so that the developer doesn't spend lots of time trying to figure out what's going on if he's making a change you don't expect.